packages/core/src/components/icon/icon.tsx

```diff
@@ -142,8 +145,16 @@ export class Icon extends AbstractPureComponent2<IconProps & Omit<React.HTMLAttr
                 className: classes,
                 title: htmlTitle,
             },
-            <svg fill={color} data-icon={icon} width={size} height={size} viewBox={viewBox}>
-                {title && <desc>{title}</desc>}
+            <svg
+                fill={color}
+                data-icon={icon}
+                width={size}
+                height={size}
+                viewBox={viewBox}
+                aria-labelledby={title ? titleId : undefined}
```

Comment:
why not just use [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label)? the `<title>` is not visible on the page anyway, so I'm not sure that `aria-labelledby` is the best approach here.

Reply:
alright, that 1st example seems quite outdated but the 2nd seems alright. still I prefer to go by what MDN says. I've verified that this is legit, according to item 5 [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby#benefits_and_drawbacks):

> The `aria-labelledby` property value can include content from elements that aren't even visible.